src/app/search/ItemSearchBar.tsx

```diff
@@ -0,0 +1,191 @@
+  mainSearchBar?: boolean;
+  /** A fake property that can be used to force the "live" query to be replaced with the one from props */
+  searchQueryVersion?: number;
+  /** The search query to fill in the input. This is used only initially, or when searchQueryVersion changes */
+  searchQuery?: string;
+  /** Children are used as optional extra action buttons only when there is a query. */
+  children?: React.ReactNode;
+  /** An optional menu of actions that can be executed on the search. Always shown. */
+  menu?: React.ReactNode;
+  instant?: boolean;
+  className?: string;
+  /** Fired whenever the query changes (already debounced) */
+  onQueryChanged: (query: string) => void;
+  /** Fired whenever the query has been cleared */
+  onClear?: () => void;
+}
+
+interface StoreProps {
+  recentSearches: Search[];
+  validateQuery: ReturnType<typeof validateQuerySelector>;
+  autocompleter: (
+    query: string,
+    caretIndex: number,
+    recentSearches: Search[],
+    includeArmory: boolean
+  ) => SearchItem[];
+}
+
+type Props = ProvidedProps & StoreProps & ThunkDispatchProp;
+
+const autoCompleterSelector = createSelector(searchConfigSelector, createAutocompleter);
+
+function mapStateToProps() {
+  let prevSearchQueryVersion: number | undefined;
+  let prevSearchQuery: string | undefined;
+
+  return (
+    state: RootState,
+    { searchQuery, searchQueryVersion }: ProvidedProps
+  ): StoreProps & { searchQuery?: string } => {
+    // This is a hack that prevents `searchQuery` from changing if `searchQueryVersion`
+    // doesn't change, so we don't trigger an update.
+    let manipulatedSearchQuery = prevSearchQuery;
+    if (searchQueryVersion === undefined) {
+      // OK, they didn't even provide searchQueryVersion, just pass through the original query
+      manipulatedSearchQuery = searchQuery;
+    } else if (searchQueryVersion !== prevSearchQueryVersion) {
+      manipulatedSearchQuery = searchQuery;
+      prevSearchQuery = searchQuery;
+      prevSearchQueryVersion = searchQueryVersion;
+    }
+
+    return {
+      recentSearches: recentSearchesSelector(state),
+      autocompleter: autoCompleterSelector(state),
+      validateQuery: validateQuerySelector(state),
+      searchQuery: manipulatedSearchQuery,
+    };
+  };
+}
+
+function ArmorySheet({ itemHash, onClose }: { itemHash: number; onClose: () => void }) {
+  return (
+    <Portal>
+      <Sheet onClose={onClose} sheetClassName={styles.armorySheet}>
+        <ClickOutsideRoot>
+          <Armory itemHash={itemHash} />
+        </ClickOutsideRoot>
+      </Sheet>
+    </Portal>
+  );
+}
+
+/**
+ * A reusable, autocompleting item search input. This is an uncontrolled input that
+ * announces its query has changed only after some delay. This is the new version of the component
+ * that offers a browser-style autocompleting search bar with history.
+ *
+ * TODO: Should this be the main search bar only, or should it also work for item picker, etc?
```

Comment:
The intention is that it works anywhere we can search items

Reply:
That comment was copy paste from the old version I am pretty sure